persistence/commons/src/main/java/org/apache/causeway/persistence/commons/integration/repository/RepositoryServiceDefault.java

```diff
@@ -99,6 +105,27 @@ public <T> T detachedEntity(final @NonNull T entity) {
         return factoryService.detachedEntity(entity);
     }
 
+    @Override
+    public <T> T execInBulk(Callable<T> callable, Class<?>... classes) {
+        try {
+            Arrays.stream(classes).forEach(aClass -> setBulkMode(aClass, Boolean.TRUE));
+            return callable.call();
+        } catch (Exception e) {
+            log.error("Error executing bulk", e);
+            return null;
+        } finally {
+            Arrays.stream(classes).forEach(aClass -> setBulkMode(aClass, Boolean.FALSE));
+        }
+    }
+
+    protected  <T extends Class> void setBulkMode(final T aClass, final Boolean bulkMode) {
```

Comment:
Let's make this `private` visibility ; if it were subclassed then it could change the semantics completely.

Reply:
That was the idea.

Reply:
Then again why would you override it?